pkg/storage/db.go

```diff
@@ -1060,3 +1067,89 @@ order by ae.ts asc
 	}
 	return nil
 }
+
+// StoreAccountOwner is an implementation of AccountOwnerStorer interface that saves account ID, its owner and champions of the account to a database
+func (db *DB) StoreAccountOwner(ctx context.Context, accountOwner domain.AccountOwner) error {
+
+	tx, err := db.sqldb.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = db.storeAccountOwner(ctx, accountOwner, tx)
+
+	if err != nil {
+		if rollbackErr := tx.Rollback(); rollbackErr != nil {
+			return errors.Wrap(rollbackErr, err.Error()) // so we don't lose the original error
+		}
+		return err
+	}
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (db *DB) storeAccountOwner(ctx context.Context, accountOwner domain.AccountOwner, tx *sql.Tx) error {
+
+	sqlStatement := fmt.Sprintf(`
+				INSERT INTO aws_account (account)
+				VALUES ($1)
+				ON CONFLICT DO NOTHING
+				`)
+	var err error
+	if _, err = tx.ExecContext(ctx, sqlStatement, accountOwner.AccountID); err != nil {
+		return err
+	}
+
+	// Insert or update details of account owner
+	if _, err = tx.ExecContext(ctx, insertPersonQuery, accountOwner.Owner.Login, accountOwner.Owner.Email, accountOwner.Owner.Name, accountOwner.Owner.Valid); err != nil {
+		return err
+	}
+
+	sqlStatement = fmt.Sprintf(`SELECT id FROM person WHERE login=$1`)
+	row := tx.QueryRowContext(ctx, sqlStatement, accountOwner.Owner.Login)
+	var personID int
+	if err := row.Scan(&personID); err != nil {
+		return err
+	}
+
+	sqlStatement = fmt.Sprintf(`SELECT id FROM aws_account WHERE account=$1`)
+	row = tx.QueryRowContext(ctx, sqlStatement, accountOwner.AccountID)
+	var accountID int
+	if err := row.Scan(&accountID); err != nil {
+		return err
+	}
+
+	sqlStatement = `INSERT INTO account_owner
+		VALUES ($1, $2)
+   		ON CONFLICT (aws_account_id) DO UPDATE SET person_id = $1, aws_account_id = $2`
+	if _, err := tx.ExecContext(ctx, sqlStatement, personID, accountID); err != nil {
+		return err
+	}
+
+	sqlStatement = fmt.Sprintf(`DELETE FROM account_champion WHERE aws_account_id=%d`, accountID)
```

Comment:
task: can we change this query to just use the `$1` style instead of using `fmt.Sprintf` to substitute the variable

Reply:
Fixed